### PR TITLE
Execute async script

### PR DESCRIPTION
**Pull request by mhanberg**
This is a replacement PR for #374.

It simply merges in master to get it up to date.

It was previously failing the build due to Selenium 2.x, but we recently dropped Selenium 2.x from our CI builds so now it is passing.

There was some worry in #374 about still wanting to support Selenium 2 due to supporting older Firefox versions, but I'm not so worried about that. The [3.0.0](https://github.com/SeleniumHQ/selenium/releases/tag/selenium-3.0.0) release seems to imply that Selenium 3 still works works older versions, but doesn't use GeckoDriver in those cases.

cc/ @michallepicki @keathley

### CODE REVIEW

**Comment by mhanberg** [comment]
@michallepicki do you mind reviewing this?

**Comment by michallepicki** [comment]
I'll try to do review it over the weekend, I actually have a use-case in my tests and want to try how using `execute_script_async` would improve my code (currently I'm callling `execute_script` checking value repeatedly)

File: lib/wallaby/experimental/selenium/webdriver_client.ex [hunk 1]
```diff
@@ -336,6 +336,17 @@ defmodule Wallaby.Experimental.Selenium.WebdriverClient do
       do: {:ok, value}
   end
 
+  @doc """
+  Executes javascript asynchronously, taking as arguments the script to execute,
```
Comment (michallepicki) [inline lib/wallaby/experimental/selenium/webdriver_client.ex]: I looked into it and I don't think it executes the script `asynchronously`, when executing the script the WebDriver client will block until the callback is called. Maybe we could reword this?

File: lib/wallaby/browser.ex [hunk 2]
```diff
@@ -410,6 +410,34 @@ defmodule Wallaby.Browser do
     parent
   end
 
+  @doc """
+  Executes javascript asynchronously, taking as arguments the script to execute,
```
Comment (michallepicki) [inline lib/wallaby/browser.ex]: I looked into it and I don't think it executes the script asynchronously, when executing the script the WebDriver client will block until the callback is called. Maybe we could reword this?

File: lib/wallaby/experimental/chrome.ex [hunk 3]
```diff
@@ -210,6 +210,21 @@ defmodule Wallaby.Experimental.Chrome do
     end
   end
 
+  @doc false
+  def execute_script_async(session_or_element, script, args \\ [], opts \\ []) do
```
Comment (michallepicki) [inline lib/wallaby/experimental/chrome.ex]: I don't really get how logging stuff works and don't use it currently, but looks the same as `execute_script` so :man_shrugging:

**Pull request merged**